5ttgen fsl failing upon mrcalc

So I've just merged `master` into `fix_strides_for_FSL`, so as to use the latest changes introduced to `5ttgen`.  I was trying to look into whether there where any issues with the strides for that one, but ran into another unrelated issue.  Just running `5ttgen fsl anat.mif 5ttout.mif` led to the following output:

```
5ttgen: 
5ttgen: Note that this script makes use of commands / algorithms that have relevant articles for citation; INCLUDING FROM EXTERNAL SOFTWARE PACKAGES. Please consult the help page (-help option) for more information.
5ttgen: 
5ttgen: Generated temporary directory: /tmp/5ttgen-tmp-P0GUP3/
Command: mrconvert anat.mif /tmp/5ttgen-tmp-P0GUP3/input.mif -stride +1,+2,+3  -quiet
5ttgen: Changing to temporary directory (/tmp/5ttgen-tmp-P0GUP3/)
Command: mrconvert input.mif T1.nii  -quiet
Command: maskfilter /usr/share/fsl/5.0/data/standard/MNI152_T1_1mm_brain_mask_dil.nii.gz dilate mni_mask.nii -npass 4  -quiet
Command: standard_space_roi T1.nii T1_preBET.nii.gz -maskMASK mni_mask.nii -roiFOV
Command: bet T1_preBET.nii.gz T1_masked.nii.gz -f 0.15 -R
Command: fast T1_masked.nii.gz
Command: run_first_all -s L_Accu,R_Accu,L_Caud,R_Caud,L_Pall,R_Pall,L_Puta,R_Puta,L_Thal,R_Thal -i T1.nii -o first
8514
Command: meshconvert first-L_Accu_first.vtk L_Accu.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve L_Accu.vtk T1_masked.nii.gz mesh2pve_L_Accu.nii  -quiet
Command: meshconvert first-R_Accu_first.vtk R_Accu.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve R_Accu.vtk T1_masked.nii.gz mesh2pve_R_Accu.nii  -quiet
Command: meshconvert first-L_Caud_first.vtk L_Caud.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve L_Caud.vtk T1_masked.nii.gz mesh2pve_L_Caud.nii  -quiet
Command: meshconvert first-R_Caud_first.vtk R_Caud.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve R_Caud.vtk T1_masked.nii.gz mesh2pve_R_Caud.nii  -quiet
Command: meshconvert first-L_Pall_first.vtk L_Pall.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve L_Pall.vtk T1_masked.nii.gz mesh2pve_L_Pall.nii  -quiet
Command: meshconvert first-R_Pall_first.vtk R_Pall.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve R_Pall.vtk T1_masked.nii.gz mesh2pve_R_Pall.nii  -quiet
Command: meshconvert first-L_Puta_first.vtk L_Puta.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve L_Puta.vtk T1_masked.nii.gz mesh2pve_L_Puta.nii  -quiet
Command: meshconvert first-R_Puta_first.vtk R_Puta.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve R_Puta.vtk T1_masked.nii.gz mesh2pve_R_Puta.nii  -quiet
Command: meshconvert first-L_Thal_first.vtk L_Thal.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve L_Thal.vtk T1_masked.nii.gz mesh2pve_L_Thal.nii  -quiet
Command: meshconvert first-R_Thal_first.vtk R_Thal.vtk -transform_first2real input.mif  -quiet
Command: mesh2pve R_Thal.vtk T1_masked.nii.gz mesh2pve_R_Thal.nii  -quiet
Command: mrmath mesh2pve_L_Accu.nii mesh2pve_R_Accu.nii mesh2pve_L_Caud.nii mesh2pve_R_Caud.nii mesh2pve_L_Pall.nii mesh2pve_R_Pall.nii mesh2pve_L_Puta.nii mesh2pve_R_Puta.nii mesh2pve_L_Thal.nii mesh2pve_R_Thal.nii sum - -quiet | mrcalc - 1.0 -min all_sgms.nii  -quiet
Command: mrthreshold T1_masked_pve_2.nii.gz - -abs 0.001 -quiet | maskfilter - connect wm_mask.mif -largest  -quiet
Command: mrconvert T1_masked_pve_0.nii.gz csf.mif  -quiet
Command: mrcalc 1 csf.mif -sub all_sgms.nii -min sgm.mif  -quiet
Command: mrcalc 1.0 csf.mif sgm.mif -add -sub T1_masked_pve_1.nii.gz T1_masked_pve_2.nii.gz -add -div multiplier.mif  -quiet
Command: mrcalc multiplier.mif -finite multiplier.mif 0.0 -if multiplier_noNAN.mif  -quiet
Command: mrcalc T1_masked_pve_1.nii.gz multiplier_noNAN.mif -mult cgm.mif  -quiet
Command: mrcalc T1_masked_pve_2.nii.gz multiplier_noNAN.mif -mult wm_mask.mif -mult wm.mif  -quiet
Command: mrcalc 0 wm.mif -min path.mif  -quiet
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] operation not supported!
mrcalc: [ERROR] exception thrown from one or more threads "loop threads"
5ttgen: [ERROR] Command failed: mrcalc 0 wm.mif -min path.mif  -quiet
```

The issue seems to be with `mrcalc`, but I've never had any issues before with using it as such (with the `-min` operator and a scalar)...  Any ideas what might be going on?

--- a/cmd/mrcalc.cpp
+++ b/cmd/mrcalc.cpp
@@ -194,6 +194,17 @@ class ThreadLocalStorage : public std::vector<ThreadLocalStorageItem> {
 
 
 
+class LoadedImage
+{
+  public:
+    LoadedImage (std::shared_ptr<Image<complex_type>>& i, const bool c) :
+        image (i),
+        image_is_complex (c) { }
+    std::shared_ptr<Image<complex_type>> image;
+    bool image_is_complex;
+};
+
+
 
 
 class StackEntry {
@@ -212,14 +223,15 @@ class StackEntry {
       auto search = image_list.find (arg);
       if (search != image_list.end()) {
         DEBUG (std::string ("image \"") + arg + "\" already loaded - re-using exising image");
-        image = search->second;
+        image = search->second.image;
+        image_is_complex = search->second.image_is_complex;
       }
       else {
         try {
           auto header = Header::open (arg);
           image_is_complex = header.datatype().is_complex();
           image.reset (new Image<complex_type> (header.get_image<complex_type>()));
-          image_list.insert (std::make_pair (arg, image));
+          image_list.insert (std::make_pair (arg, LoadedImage (image, image_is_complex)));
         }
         catch (Exception) {
           std::string a = lowercase (arg);
@@ -245,12 +257,12 @@ class StackEntry {
 
     bool is_complex () const;
 
-    static std::map<std::string, std::shared_ptr<Image<complex_type>>> image_list;
+    static std::map<std::string, LoadedImage> image_list;
 
     Chunk& evaluate (ThreadLocalStorage& storage) const;
 };
 
-std::map<std::string, std::shared_ptr<Image<complex_type>>> StackEntry::image_list;
+std::map<std::string, LoadedImage> StackEntry::image_list;
 
 
 class Evaluator
@@ -486,7 +498,7 @@ void unary_operation (const std::string& operation_name, std::vector<StackEntry>
       throw Exception ("operation \"" + operation_name + "\" not supported for data type supplied");
     }
   }
-};
+}
 
 
 

